src/title.js

```diff
@@ -0,0 +1,437 @@
+				if ( allowedToBeTitle.has( rootChild.name ) ) {
+					const position = model.createPositionBefore( rootChild );
+					const title = writer.createElement( 'title' );
+
+					writer.insert( title, position );
+					writer.append( rootChild, title );
+					writer.rename( rootChild, 'title-content' );
+
+					// After changing element to the title it has to be filtered out from disallowed attributes.
+					model.schema.removeDisallowedAttributes( Array.from( rootChild.getChildren() ), writer );
+
+				// If the first element cannot be a title but title is already in the root
+				// than move the first element after a title.
+				// It may happen e.g. when an image has been dropped before the title element.
+				} else if ( title ) {
+					writer.move( writer.createRangeOn( rootChild ), title, 'after' );
+
+				// If there is no title or any element that could be a title then create an empty title.
+				} else {
+					writer.insertElement( 'title', modelRoot );
+					writer.insertElement( 'title-content', modelRoot.getChild( 0 ) );
+				}
+
+				hasChanged = true;
+
+			// If there is a title somewhere in the content.
+			} else if ( index > 0 && rootChild.is( 'title' ) ) {
+				// Rename it to a paragraph if it has a content.
+				if ( model.hasContent( rootChild ) ) {
+					writer.rename( rootChild, 'paragraph' );
+					writer.unwrap( rootChild.getChild( 0 ) );
+				// Or remove if it's empty (it's created as a result of splitting parents by insertContent method).
+				} else {
+					writer.remove( rootChild );
+				}
+
+				hasChanged = true;
+			}
+
+			index++;
+		}
+
+		// Attach `Body` placeholder when there is no element after a title.
+		if ( modelRoot.childCount < 2 ) {
+			this._bodyPlaceholder = writer.createElement( 'paragraph' );
+			writer.insert( this._bodyPlaceholder, modelRoot, 1 );
+			hasChanged = true;
+		}
+
+		return hasChanged;
+	}
+
+	/**
+	 * Prevents editor from adding extra paragraphs after pasting to the title element
+	 * or pressing an enter in the title element.
+	 *
+	 * @private
+	 * @param {module:engine/model/writer~Writer} writer
+	 * @returns {Boolean}
+	 */
+	_preventExtraParagraphing( writer ) {
+		const root = this.editor.model.document.getRoot();
+		const placeholder = this._bodyPlaceholder;
+
+		const shouldDeleteLastParagraph = (
+			placeholder && placeholder.is( 'paragraph' ) && placeholder.childCount === 0 &&
+			root.childCount > 2 && root.getChild( root.childCount - 1 ) === placeholder
+		);
+
+		if ( shouldDeleteLastParagraph ) {
+			this._bodyPlaceholder = null;
+			writer.remove( placeholder );
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Attaches `Title` and `Body` placeholders to the title and/or content.
+	 *
+	 * @private
+	 */
+	_attachPlaceholders() {
+		const editor = this.editor;
+		const view = editor.editing.view;
+		const viewRoot = view.document.getRoot();
+
+		const bodyPlaceholder = editor.config.get( 'placeholder' ) || 'Body';
```

Comment:
I'll check this but I feel that we should make those strings (`'Title'` & `'Body'`) translate-able.

Reply:
No doubts. I'll change it.